ui/shared/Input.qml

```diff
@@ -108,47 +109,57 @@ Item {
             anchors.verticalCenter: parent.verticalCenter
             anchors.right: parent.right
             anchors.rightMargin: 8
-            sourceComponent: copyToClipboardComponent
-        }
-
-        Component {
-            id: copyToClipboardComponent
+            sourceComponent: Component {
+                Item {
+                    width: copyBtn.width
+                    height: copyBtn.height
 
-            Item {
-                width: copyBtn.width
-                height: copyBtn.height
-
-                Timer {
-                    id: timer
-                }
+                    Timer {
+                        id: timer
+                    }
 
-                StyledButton {
-                    id: copyBtn
-                    //% "Copy"
-                    label: qsTrId("copy-to-clipboard")
-                    height: 28
-                    textSize: 12
-                    btnBorderColor: Style.current.blue
-                    btnBorderWidth: 1
-                    onClicked: {
-                        chatsModel.copyToClipboard(inputValue.text)
-                        //% "Copied"
-                        copyBtn.label = qsTrId("sharing-copied-to-clipboard")
-                        timer.setTimeout(function(){
-                            //% "Copy"
-                            copyBtn.label = qsTrId("copy-to-clipboard")
-                        }, 2000);
+                    StyledButton {
+                        property bool copied: false
+                        id: copyBtn
+                        label: {
+                            if (copied) {
+                                return inputBox.copyToClipboard ?
+                                            //% "Copied"
+                                            qsTrId("sharing-copied-to-clipboard")  :
+                                            qsTr("Pasted")
+                            }
+                            return inputBox.copyToClipboard ?
+                                        //% "Copy"
+                                        qsTrId("copy-to-clipboard") :
+                                        qsTr("Paste")
+
+                        }
+                        height: 28
+                        textSize: 12
+                        btnBorderColor: Style.current.blue
+                        btnBorderWidth: 1
+                        onClicked: {
+                            if (inputBox.copyToClipboard) {
+                                chatsModel.copyToClipboard(inputValue.text)
+                            } else {
+                                if (inputValue.canPaste) {
+                                    inputValue.paste()
+                                }
+                            }
+
+                            copyBtn.copied = true
+                            timer.setTimeout(function() {
+                                copyBtn.copied = false
+                            }, 2000);
```

Comment:
We have a lot of different timeout values. I wonder if we should add this value to Constants.qml

Reply:
We can, but I just checked and it's the only time we use 2000. I'm not sure if we should have a standard or if it doesn't matter